pkgs/applications/version-management/stgit/default.nix

```diff
@@ -40,24 +45,20 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace Documentation/texi.xsl \
       --replace http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd \
                 ${docbook_xml_dtd_45}/xml/dtd/docbook/docbookx.dtd
-
-    cat > stgit/_version.py <<EOF
-    __version__ = "${version}"
-    EOF
   '';
 
   makeFlags = [
     "prefix=${placeholder "out"}"
     "MAN_BASE_URL=${placeholder "out"}/share/man"
     "XMLTO_EXTRA=--skip-validation"
+    "PERL_PATH=${perl}/bin/perl"
   ];
 
-  buildFlags = [ "all" "doc" ];
+  buildPhase = [ "make" "all" ] ++ makeFlags;
 
-  checkTarget = "test";
-  checkFlags = [ "PERL_PATH=${perl}/bin/perl" ];
+  checkPhase = [ "make" "test" ] ++ makeFlags;
```

Comment:
This should be checkTarget

Reply:
I'm not sure how that would work. This now uses `rustPlatform.buildRustPackage` which ignores checkTargets. Instead, it just runs the Rust test then, which are very different than the integration tests behind `make test`.

For a Rust based program that use a Makefiles, the nixpkgs docs write the following: 

> Some packages may use custom scripts for building/installing, e.g. with a Makefile. In these cases, it’s recommended to override the buildPhase/installPhase/checkPhase. 

(from https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo, 17.31.1.6. Custom build/install-procedures)

Using checkTarget that automatically uses the makeFlags sounds cleaner, I'm just now sure how I would use that with Rust.

Reply:
As this is the recommended way to package Rust programs with custom build processes, I'll stick with checkPhase. But I updated it to be a multiline string as you noted.